File: schemaorggo/audience.go
package schemaorggo

import "encoding/json"

// Audience see : https://schema.org/Audience
type Audience struct {
	Intangible

	typeContext

	// AudienceType see : https://schema.org/audienceType
	// The target group associated with a given audience (e.g. veterans, car owners, musicians, etc.).
	// types : Text
	AudienceType []string `json:"audienceType,omitempty"`

	// GeographicArea see : https://schema.org/geographicArea
	// The geographic area associated with the audience.
	// types : AdministrativeArea
	GeographicArea []*AdministrativeArea `json:"geographicArea,omitempty"`
}

func (v Audience) IntoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.Intangible.IntoMap(intop)

	into := *intop

	{
		var value interface{} = v.AudienceType
		if len(v.AudienceType) == 1 {
			value = v.AudienceType[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["audienceType"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.GeographicArea
		if len(v.GeographicArea) == 1 {
			value = v.GeographicArea[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["geographicArea"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v Audience) AsMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.IntoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "Audience"

	return data, nil
}

func (v Audience) MarshalJSON() ([]byte, error) {
	data, err := v.AsMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/audience.php
<?php

class Audience extends Intangible implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'Audience';
	
	/**
	 * The target group associated with a given audience (e.g. veterans, car owners, musicians, etc.).
	 * see : https://schema.org/audienceType
	 * @var string|string[]
	 */
	public var $audience_type;
	
	/**
	 * The geographic area associated with the audience.
	 * see : https://schema.org/geographicArea
	 * @var \AdministrativeArea|\AdministrativeArea[]
	 */
	public var $geographic_area;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'Audience'
		);
		
		$serialized = so_json_serialize( $this->audience_type );
		if ( ! empty( $serialized ) ) {
			$out['audienceType'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->geographic_area );
		if ( ! empty( $serialized ) ) {
			$out['geographicArea'] = $serialized;
		}
		
		return $out;
	}
}
